src/entity-manager/EntityManager.ts

```diff
@@ -178,7 +180,9 @@ export class EntityManager extends BaseEntityManager {
         } else {
             // todo: throw exception if constructor in target is not set
             if (target instanceof Array) {
-                return this.getRepository<Entity[]>(target[0].constructor).remove(entity as Entity[], options);
+                return Promise.all(target.map((t, i) => {
```

Comment:
I don't like short variables like `t` and `i`, and prefer to name them full name, e.g. `target` or `targetItem`, or simply `item` and `index`, instead of `t` and `i`. But its not that important, I can do it by myself next time I face this code. Thank you again